src/server/game/DungeonFinding/LFGMgr.cpp

```diff
@@ -710,6 +713,12 @@ void LFGMgr::UpdateRoleCheck(ObjectGuid gguid, ObjectGuid guid /* = ObjectGuid::
     // Sanitize input roles
     roles &= PLAYER_ROLE_ANY;
 
+    if (guid) {
+        Player* player = ObjectAccessor::FindPlayer(guid);
+        if (!player) return;
+        roles = FilterClassRoles(player, roles);
+    }
+
```

Comment:
```suggestion
    if (guid)
    {
        if (Player* player = ObjectAccessor::FindPlayer(guid))
            roles = FilterClassRoles(player, roles);
    }

```

Reply:
This one is functionally different, we need to exit the function if the player doesn't exist, not allow it to continue without filtering

Reply:
still, the formatting needs to be changed for these lines

Reply:
sorry, completely forgot about this pull. i'll fix it tomorrow!